src/gnatss/ops/solve.py

```diff
@@ -287,12 +409,35 @@ def solve_transponder_locations(
 
 @numba.njit(cache=True)
 def perform_solve(
-    data_inputs,
-    transponders_mean_sv,
-    transponders_xyz,
-    transponders_delay,
-    travel_times_variance,
+    data_inputs: NumbaList,
+    transponders_mean_sv: NDArray[Shape["*"], Float64],
+    transponders_xyz: NDArray[Shape["*, 3"], Float64],
+    transponders_delay: NDArray[Shape["*"], Float64],
+    travel_times_variance: float,
 ):
+    """
+    Perform the solve for the given data inputs and transponder information.
+
+    Parameters
+    ----------
+    data_inputs : NumbaList
+        A list of tuples, where each tuple contains the transmit and reply positions,
+        the GPS covariance matrix, and the observed delays for a single ping.
+    transponders_mean_sv : (N,) ndarray
+        The mean signal velocity for each transponder.
+    transponders_xyz : (N,3) ndarray
+        The XYZ positions of each transponder.
+    transponders_delay : (N,) ndarray
+        The internal delay for each transponder.
+    travel_times_variance : float
+        The variance in travel times
+
```

Comment:
nit: either end all or none with a period (for a single sentence usually no period is fine)